### ISSUE TITLE
”As a writer“ doesn't detect series-level metadata [People Page]

### ISSUE THREAD

**Issue by newyorker123** [Kareadita/Kavita#3370]
### What happened?

On the People page, the "As a role" section can't detect the role if it exists only at the series level, i.e., not at the file level.

Take PDF as an example: since PDF files don't have per-file ComicInfo, I manually add the writer at the series level using the Kavita UI. The issue is that the series is listed under "Known for" but not under "As a Writer."

![屏幕截图 2024-11-15 221149](https://github.com/user-attachments/assets/d2f2b665-0e6d-4ed0-9abf-7e375afb78ae)

![屏幕截图 2024-11-15 221211](https://github.com/user-attachments/assets/92978f4f-d16f-4e81-8937-13de10e35376)

### What did you expect?

I understand why this happens. It’s because "Known for" lists series, whereas "As roles" lists files. One simple way to address this is to have "As a role" behave like "Known for," i.e., list series instead of files.

However, it’s up to the developers to decide if this is the intended behavior.

### Kavita Version Number - If you don not see your version number listed, please update Kavita and see if your issue still persists.

Nightly Testing Branch

### What operating system is Kavita being hosted from?

None

### If the issue is being seen on Desktop, what OS are you running where you see the issue?

None

### If the issue is being seen on Mobile, what OS are you running where you see the issue?

None

**Comment by majora2007** [comment]
This isn't a bug because pulling the individual chapters the artist is a role for, is chapter level. Yet you only have this person on a Series, hence there is no relation between the two.

**Comment by newyorker123** [comment]
When I add a writer or penciler to a series, what I actually mean is that they are responsible for every volume or chapter in the series. Ideally, I’d like to tag every file, that's what I do for cbz. But formats like PDF or EPUB aren’t very flexible, so the only option is to do it through the UI.

Another workaround could be adding a bulk metadata editing feature that allows you to assign a person to all volumes or chapters at once. Otherwise, it becomes too tedious to manually tag each file, especially for series with 10 or more files.

**Comment by majora2007** [comment]
Yeah and I totally agree with the sentiment. I had originally planned for issue-level people tagging to recalculate at the series level, however, I'm not sure if we can do series -> issues. I feel like for Comic users that might be a problem.

**Comment by Cducharme84** [comment]
I’m not sure how feasible it would be for execution but presuming these entries are manually set since they’re not file level perhaps any author or artist in series level spot but not within chapter/issue/volume level then they could be assumed full ownership of each entry in the series.

Since comic users tend to tag at a file level to get issue level summaries and other data, especially with the CV library type, I think that assumption is unlikely to have much side effect and maybe will help those who follow smaller creator owned properties where that is still usually a very safe assumption like with manga and like manga there’s not issue level metadata on common sources.
